demos/999-storage-import-vhd-blob/azuredeploy.json

```diff
@@ -0,0 +1,169 @@
+                        "forceUpdateTag": "[parameters('forceUpdateTag')]"
+                    },
+                    "resources": [
+                        {
+                            "type": "Microsoft.ManagedIdentity/userAssignedIdentities",
+                            "apiVersion": "2018-11-30",
+                            "name": "[variables('userAssignedIdentityName')]",
+                            "location": "[parameters('location')]"
+                        },
+                        {
+                            "type": "Microsoft.Authorization/roleAssignments",
+                            "apiVersion": "2018-09-01-preview",
+                            "name": "[variables('roleAssignmentName')]",
+                            "dependsOn": [
+                                "[resourceId('Microsoft.ManagedIdentity/userAssignedIdentities', variables('userAssignedIdentityName'))]"
+                            ],
+                            "properties": {
+                                "roleDefinitionId": "[resourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c')]",
+                                "principalId": "[reference(resourceId('Microsoft.ManagedIdentity/userAssignedIdentities', variables('userAssignedIdentityName')), '2015-08-31-preview').principalId]",
+                                "scope": "[resourceGroup().id]",
+                                "principalType": "ServicePrincipal"
+                            }
+                        },
+                        {
+                            "type": "Microsoft.Resources/deploymentScripts",
+                            "apiVersion": "2020-10-01",
+                            "name": "[parameters('deploymentScriptName')]",
+                            "location": "[parameters('location')]",
+                            "dependsOn": [
+                                "[resourceId('Microsoft.Authorization/roleAssignments', variables('roleAssignmentName'))]"
+                            ],
+                            "identity": {
+                                "type": "UserAssigned",
+                                "userAssignedIdentities": {
+                                    "[resourceId('Microsoft.ManagedIdentity/userAssignedIdentities',variables('userAssignedIdentityName'))]": {}
+                                }
+                            },
+                            "kind": "AzurePowerShell",
+                            "properties": {
+                                "azPowerShellVersion": "5.5",
+                                "cleanupPreference": "OnSuccess",
+                                "arguments": "[concat('-Source \\\"',variables('uriVHDBlobSasToken'),'\\\"',' ','-Destination \\\"',variables('uriWritableStorageAccountBlobContainerSasToken'),'\\\"')]",
+                                "environmentVariables": [
```

Comment:
this seems an odd pattern - to use env vars for some settings and then params for others?  Why the difference?

Reply:
:) thanks, It is cleaner now, kept the arguments.

For this an the other points above, it is because i have been reusing my favorite code snippet to add deploymentScript (with identify, resource group in the env,... but unnecessary here).